notes/zrq/20201022-01-gaia-zeppelin.txt

```diff
@@ -0,0 +1,684 @@
+    >   	at org.apache.zeppelin.interpreter.remote.RemoteClientFactory.create(RemoteClientFactory.java:54)
+    >   	at org.apache.zeppelin.interpreter.remote.RemoteClientFactory.create(RemoteClientFactory.java:34)
+    >   	at org.apache.zeppelin.shaded.org.apache.commons.pool2.BasePooledObjectFactory.makeObject(BasePooledObjectFactory.java:60)
+    >   	at org.apache.zeppelin.shaded.org.apache.commons.pool2.impl.GenericObjectPool.create(GenericObjectPool.java:861)
+    >   	at org.apache.zeppelin.shaded.org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:435)
+    >   	at org.apache.zeppelin.interpreter.remote.PooledRemoteClient.getClient(PooledRemoteClient.java:54)
+    >   	at org.apache.zeppelin.interpreter.remote.PooledRemoteClient.callRemoteFunction(PooledRemoteClient.java:92)
+    >   	... 4 more
+    >   Caused by: org.apache.zeppelin.shaded.org.apache.thrift.transport.TTransportException: java.net.UnknownHostException: zeppelin-server.default.svc
+    >   	at org.apache.zeppelin.shaded.org.apache.thrift.transport.TSocket.open(TSocket.java:226)
+    >   	at org.apache.zeppelin.interpreter.remote.RemoteInterpreterEventClient.lambda$new$0(RemoteInterpreterEventClient.java:71)
+    >   	... 11 more
+    >   Caused by: java.net.UnknownHostException: zeppelin-server.default.svc
+    >   	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:184)
+    >   	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
+    >   	at java.net.Socket.connect(Socket.java:607)
+    >   	at org.apache.zeppelin.shaded.org.apache.thrift.transport.TSocket.open(TSocket.java:221)
+    >   	... 12 more
+    >    INFO [2020-10-22 16:40:28,112] ({Shutdown-Thread} RemoteInterpreterServer.java[lambda$shutdown$0]:274) - Shutting down...
+    >    INFO [2020-10-22 16:40:28,113] ({Shutdown-Thread} SchedulerFactory.java[<init>]:62) - Scheduler Thread Pool Size: 100
+    >    INFO [2020-10-22 16:40:28,113] ({Shutdown-Thread} SchedulerFactory.java[destroy]:67) - Destroy all executors
+
+
+    Found possible cause, the config property ZEPPELIN_K8S_SERVICE_NAME defaults to "zeppelin-server"
+
+        zeppelin-interpreter/src/main/java/org/apache/zeppelin/conf/ZeppelinConfiguration.java
+
+        public enum ConfVars {
+            ....
+            ZEPPELIN_K8S_SERVICE_NAME("zeppelin.k8s.service.name", "zeppelin-server"),
+            ....
+            }
+
+    After a day to find it ..
+    Add a new value in our config map to set ZEPPELIN_K8S_SERVICE_NAME.
+    Works :-)
+
+
+# -----------------------------------------------------
+# -----------------------------------------------------
+# Mount the share in our Spark workers and test.
+#[user@zeppelin]
+
+    # --------------------------------
+
+    %spark.conf
+
+    spark.executor.instances 10
+
+    spark.kubernetes.executor.volumes.persistentVolumeClaim.gaia-dr2.mount.path        /gaia-dr2
```

Comment:
Woud we add these mount configuration parameters in the Zeppelin/Spark conf so that this extra cell isn't required?

Reply:
Yes, absolutely.
At this stage I'm just proving that the mount works.
In fact, in the final public system, we might protect access to prevent users from changing some of these settings.

Reply:
Ok, good stuff